### PR TITLE
deps(go-oscal): upgrade to v0.3.0

### PR DESCRIPTION
**Pull request by mike-winberry**
- Updated all go-oscal references to use pointers (where appropriate). 
- Update to go-oscal v0.3.0

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/defenseunicorns/lula
 go 1.22.1
 
 require (
-	github.com/defenseunicorns/go-oscal v0.2.5
+	github.com/defenseunicorns/go-oscal v0.2.6-0.20240319175731-6df16e571947
```
Comment (mike-winberry) [inline go.mod]: Would need to update to release after merging and releasing https://github.com/defenseunicorns/go-oscal/pull/170

File: src/cmd/validate/validate.go [hunk 2]
```diff
@@ -135,8 +135,8 @@ func ValidateOnCompDef(compDef oscalTypes_1_1_2.ComponentDefinition) (map[string
 	findings := make(map[string]oscalTypes_1_1_2.Finding)
 	observations := make([]oscalTypes_1_1_2.Observation, 0)
 
-	for _, component := range compDef.Components {
-		for _, controlImplementation := range component.ControlImplementations {
+	for _, component := range *compDef.Components {
```
Comment (mike-winberry) [inline src/cmd/validate/validate.go:145]: Needs nil check

File: src/cmd/validate/validate.go [hunk 3]
```diff
@@ -135,8 +135,8 @@ func ValidateOnCompDef(compDef oscalTypes_1_1_2.ComponentDefinition) (map[string
...
+		for _, controlImplementation := range *component.ControlImplementations {
```
Comment (mike-winberry) [inline src/cmd/validate/validate.go]: nil check

**Review by brandtkeller — APPROVED** [review]
Generated `assessment-results` outputs between `v0.1.0` and this branch are consistent. Glad to have this change in before we start the heavy lifting of generation.

**Review by meganwolf0 — APPROVED** [review]

**Pull request merged**